src/main/java/org/jboss/snowdrop/licenses/provider/ExternalLicenseProvider.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2016-2017 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.snowdrop.licenses.provider;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.type.CollectionType;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import org.jboss.snowdrop.licenses.LicensesGeneratorException;
+import org.jboss.snowdrop.licenses.xml.LicenseElement;
+
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+/**
+ * @author Michal Szynkiewicz, michal.l.szynkiewicz@gmail.com
+ * <br>
+ * Date: 10/20/17
+ */
+public class ExternalLicenseProvider {
+
+    private final Logger logger = Logger.getLogger(ExternalLicenseProvider.class.getSimpleName());
+
+    private final WebTarget webTarget;
+
+    public ExternalLicenseProvider(String licenseServerUrl) {
+        webTarget = ClientBuilder.newClient().target(licenseServerUrl);
+    }
+
+    public Optional<Set<LicenseElement>> getLicenses(String gav) {
```

Comment:
Why not return an empty set in case of an error same as if service doesn't know about the gav?

Reply:
Done